python/pyarrow/tests/test_plasma.py

```diff
@@ -0,0 +1,685 @@
+    # If a reference buffer was provided, check that it is the same as well.
+    if memory_buffer is not None:
+        assert plasma.buffers_equal(memory_buffer, client1_buff)
+    # If reference metadata was provided, check that it is the same as well.
+    if metadata is not None:
+        assert plasma.buffers_equal(metadata, client1_metadata)
+
+
+def start_plasma_store(plasma_store_memory=DEFAULT_PLASMA_STORE_MEMORY,
+                       use_valgrind=False, use_profiler=False,
+                       stdout_file=None, stderr_file=None):
+    """Start a plasma store process.
+    Args:
+        use_valgrind (bool): True if the plasma store should be started inside
+            of valgrind. If this is True, use_profiler must be False.
+        use_profiler (bool): True if the plasma store should be started inside
+            a profiler. If this is True, use_valgrind must be False.
+        stdout_file: A file handle opened for writing to redirect stdout to. If
+            no redirection should happen, then this should be None.
+        stderr_file: A file handle opened for writing to redirect stderr to. If
+            no redirection should happen, then this should be None.
+    Return:
+        A tuple of the name of the plasma store socket and the process ID of
+            the plasma store process.
+    """
+    if use_valgrind and use_profiler:
+        raise Exception("Cannot use valgrind and profiler at the same time.")
+    plasma_store_executable = os.path.join(pa.__path__[0], "plasma_store")
+    plasma_store_name = "/tmp/plasma_store{}".format(random_name())
+    command = [plasma_store_executable,
+               "-s", plasma_store_name,
+               "-m", str(plasma_store_memory)]
+    if use_valgrind:
+        pid = subprocess.Popen(["valgrind",
+                                "--track-origins=yes",
+                                "--leak-check=full",
+                                "--show-leak-kinds=all",
+                                "--leak-check-heuristics=stdstring",
+                                "--error-exitcode=1"] + command,
+                               stdout=stdout_file, stderr=stderr_file)
+        time.sleep(1.0)
+    elif use_profiler:
+        pid = subprocess.Popen(["valgrind", "--tool=callgrind"] + command,
+                               stdout=stdout_file, stderr=stderr_file)
+        time.sleep(1.0)
+    else:
+        pid = subprocess.Popen(command, stdout=stdout_file, stderr=stderr_file)
+        time.sleep(0.1)
+    return plasma_store_name, pid
+
+
+@pytest.mark.plasma
+class TestPlasmaClient(object):
+
+    def setup_method(self, test_method):
+        import pyarrow.plasma as plasma
+        # Start Plasma store.
+        plasma_store_name, self.p = start_plasma_store(
+            use_valgrind=os.getenv("PLASMA_VALGRIND") == "1")
+        # Connect to Plasma.
+        self.plasma_client = plasma.PlasmaClient()
+        self.plasma_client.connect(plasma_store_name, "", 64)
```

Comment:
What do you think about adding a top level function `pyarrow.plasma.connect` that returns a fully-baked client? This API is a slight rough edge